libraries/abstractions/common_io/test/test_iot_uart.c

```diff
@@ -390,16 +393,16 @@ TEST( TEST_IOT_UART, AFQP_IotUARTWriteAsyncReadAsyncLoopbackTest )
 
     if( TEST_PROTECT() )
     {
+        lRead = iot_uart_read_async( xUartHandle, cpBufferRead, testIotUART_BUFFER_LENGTH );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lRead );
+
         lWrite = iot_uart_write_async( xUartHandle, cpBuffer, testIotUART_BUFFER_LENGTH );
         TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lWrite );
 
         lIoctl = iot_uart_ioctl( xUartHandle, eGetTxNoOfbytes, &lTransferAmount );
         TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
         TEST_ASSERT_GREATER_THAN( 0, lTransferAmount );
 
-        lRead = iot_uart_read_async( xUartHandle, cpBufferRead, testIotUART_BUFFER_LENGTH );
```

Comment:
I see your point here. But I suppose it should work with write-then-read in an async way, even there is no hardware buffer for RX, assuming the full-duplex being supported. 

Because write_async should return and call read_async immediately. I might be wrong and I will confirm with team. Please also share your thoughts.

Reply:
I can tell you from experience that write -> read can not be expected to succeed. No matter the closeness of the calls, the write will in most systems start without a delay and as there will always be processing time to get into the read call and set it up, a success is not guaranteed (er even likely).

This is of course very device specific, depending on they underlying HW and driver implementations. All devices should however be able to support async read -> write.

Reply:
What you said makes sense to me.